### PR TITLE
UP-4489 - Removes makefile

### PR DESCRIPTION
**Pull request by uschischroder**

### CODE REVIEW

File: .github/workflows/ci-base.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    branches: [ master ]
+    paths:
+      - packages/base/**
+      - ./.github/workflows/ci-base.yml
+  pull_request:
+    branches: [ master ]
+    paths:
+      - packages/base/**
+      - ./.github/workflows/ci-base.yml
+jobs:
```
Comment (gustavoaz7) [inline .github/workflows/ci-base.yml:14]: [same for all workflows]
```suggestion
      - ./.github/workflows/ci-base.yml

jobs:
```

File: .github/workflows/ci-base.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ci-base:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    defaults:
+      run:
+        working-directory: "./packages/base"
+    env:
+      NPM_CI: true
```
Comment (gustavoaz7) [inline .github/workflows/ci-base.yml:22]: [same for all workflows]
```suggestion
```

File: .github/workflows/ci-browser.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    paths:
+      - packages/browser/**
+      - ./.github/workflows/ci-browser.yml
+  pull_request:
+    branches: [ master ]
+    paths:
+      - packages/browser/**
+      - ./.github/workflows/ci-browser.yml
+jobs:
+  ci-base:
```
Comment (gustavoaz7) [inline .github/workflows/ci-browser.yml:15]:
```suggestion
  ci-browser:
```

File: .github/workflows/ci-node.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    paths:
+      - packages/node/**
+      - ./.github/workflows/ci-node.yml
+  pull_request:
+    branches: [ master ]
+    paths:
+      - packages/node/**
+      - ./.github/workflows/ci-node.yml
+jobs:
+  ci-base:
```
Comment (gustavoaz7) [inline .github/workflows/ci-node.yml:15]:
```suggestion
  ci-node:
```

File: .github/workflows/ci-base.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Clean
+        run: npm run clean
+      - name: Install node modules
```
Comment (gustavoaz7) [inline .github/workflows/ci-base.yml:32]: [IMO, aka opinionated suggestion]
```suggestion
          node-version: 16
      - name: Clean
        run: npm run clean
      - name: Install
```

**Review by gustavoaz7 — CHANGES_REQUESTED** [review]
- `node` and `browser` CIs require bootstrap in order to access `base` lib
- `bump-version` should not be defined in packages scripts

File: .github/workflows/ci-browser.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+name: ci-browser
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - packages/browser/**
+      - ./.github/workflows/ci-browser.yml
```
Comment (enjoythelive1) [inline .github/workflows/ci-browser.yml:8]:
```suggestion
      - packages/base/**
      - packages/browser/**
      - ./.github/workflows/ci-browser.yml
```

base is a dependency of browser. Meaning that changes there could make this build fail.

File: .github/workflows/ci-browser.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  pull_request:
+    branches: [ master ]
+    paths:
+      - packages/browser/**
```
Comment (enjoythelive1) [inline .github/workflows/ci-browser.yml:12]:
```suggestion
      - packages/base/**
      - packages/browser/**
```

File: .github/workflows/ci-node.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
+name: ci-node
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - packages/node/**
```
Comment (enjoythelive1) [inline .github/workflows/ci-node.yml:7]:
```suggestion
      - packages/base/**
      - packages/node/**
```

File: .github/workflows/ci-node.yml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      - ./.github/workflows/ci-node.yml
+  pull_request:
+    branches: [ master ]
+    paths:
+      - packages/node/**
```
Comment (enjoythelive1) [inline .github/workflows/ci-node.yml:12]:
```suggestion
      - packages/base/**
      - packages/node/**
```

File: packages/base/package.json [hunk 10]
```diff
@@ -7,11 +7,11 @@
     "build:stage": "ENV=staging rollup --config rollup.config.js",
     "build:mock": "ENV=mock rollup --config rollup.config.js",
     "lint": "eslint --ext .js,.ts .",
+    "clean": "rm -rf ./dist",
     "prepublishOnly": "make ci",
     "test": "jest --passWithNoTests",
     "test:watch": "npm test -- --watch",
-    "test:coverage": "npm test -- --coverage",
-    "bump-version": "lerna version --no-git-tag-version --no-push",
+    "test:coverage": "npm test -- --ci --json --coverage --testLocationInResults --outputFile=coverage/report.json",    "bump-version": "lerna version --no-git-tag-version --no-push",
```
Comment (enjoythelive1) [inline packages/base/package.json:14]:
```suggestion
    "test:coverage": "npm test -- --ci --json --coverage --testLocationInResults --outputFile=coverage/report.json",
    "bump-version": "lerna version --no-git-tag-version --no-push",
```

File: packages/browser/package.json [hunk 11]
```diff
@@ -7,11 +7,11 @@
     "build:stage": "ENV=staging rollup --config rollup.config.js",
     "build:mock": "ENV=mock rollup --config rollup.config.js",
     "lint": "eslint --ext .js,.ts .",
+    "clean": "rm -rf ./dist",
     "prepublishOnly": "make ci",
     "test": "jest --passWithNoTests",
     "test:watch": "npm test -- --watch",
-    "test:coverage": "npm test -- --coverage",
-    "makeTypeDeclarations": "tsc -d --skipLibCheck --declarationDir dist/types --emitDeclarationOnly index.ts",
+    "test:coverage": "npm test -- --ci --json --coverage --testLocationInResults --outputFile=coverage/report.json",    "makeTypeDeclarations": "tsc -d --skipLibCheck --declarationDir dist/types --emitDeclarationOnly index.ts",
```
Comment (enjoythelive1) [inline packages/browser/package.json:14]:
```suggestion
    "test:coverage": "npm test -- --ci --json --coverage --testLocationInResults --outputFile=coverage/report.json",
    "makeTypeDeclarations": "tsc -d --skipLibCheck --declarationDir dist/types --emitDeclarationOnly index.ts",
```

File: packages/node/package.json [hunk 12]
```diff
@@ -7,11 +7,11 @@
     "build:stage": "ENV=staging rollup --config rollup.config.js",
     "build:mock": "ENV=mock rollup --config rollup.config.js",
     "lint": "eslint --ext .js,.ts .",
+    "clean": "rm -rf ./dist",
     "prepublishOnly": "make ci",
     "test": "jest --passWithNoTests",
     "test:watch": "npm test -- --watch",
-    "test:coverage": "npm test -- --coverage",
-    "bump-version": "lerna version --no-git-tag-version --no-push",
+    "test:coverage": "npm test -- --ci --json --coverage --testLocationInResults --outputFile=coverage/report.json",    "bump-version": "lerna version --no-git-tag-version --no-push",
```
Comment (enjoythelive1) [inline packages/node/package.json:14]:
```suggestion
    "test:coverage": "npm test -- --ci --json --coverage --testLocationInResults --outputFile=coverage/report.json",
    "bump-version": "lerna version --no-git-tag-version --no-push",
```

**Review by enjoythelive1 — CHANGES_REQUESTED** [review]

**Review by enjoythelive1 — APPROVED** [review]

**Comment by gustavoaz7** [comment]
> * `bump-version` should not be defined in packages scripts

Please address previous bullet point.
CI is currently failing

**Review by Drafter500 — APPROVED** [review]

File: .github/workflows/ci-node.yml (hunk 8 above)
Comment (enjoythelive1) [reply .github/workflows/ci-node.yml:7]: > 

:arrow_up:

File: .github/workflows/ci-node.yml (hunk 9 above)
Comment (enjoythelive1) [reply .github/workflows/ci-node.yml:13]: :arrow_up:

File: .github/workflows/ci-node.yml (hunk 8 above)
Comment (uschischroder) [reply .github/workflows/ci-node.yml:7]: it's already added

**Review by gustavoaz7 — CHANGES_REQUESTED** [review]
Makefiles have been removed but they are still in use
`"prepublishOnly": "make ci",`

I think it's worth putting some effort into better covering the codebase with test instead of decreasing our threshold

File: .github/workflows/ci-browser.yml (hunk 7 above)
Comment (enjoythelive1) [reply .github/workflows/ci-browser.yml:13]: :arrow_up:

**Comment by uschischroder** [comment]
> Makefiles have been removed but they are still in use `"prepublishOnly": "make ci",`
> 
> I think it's worth putting some effort into better covering the codebase with test instead of decreasing our threshold

It will be done in a separate ticket

**Review by enjoythelive1 — APPROVED** [review]

File: packages/base/package.json [hunk 13]
```diff
@@ -8,7 +8,7 @@
     "build:mock": "ENV=mock rollup --config rollup.config.js",
     "lint": "eslint --ext .js,.ts .",
     "clean": "rm -rf ./dist",
-    "prepublishOnly": "make ci",
+    "prepublishOnly": "npm run ci",
```
Comment (gustavoaz7) [inline packages/base/package.json:11]: There is no `ci` script
_[same for all packages]_

**Review by gustavoaz7 — CHANGES_REQUESTED** [review]

File: packages/base/package.json (hunk 13 above)
Comment (uschischroder) [reply packages/base/package.json:11]: sorry, my bad...adding it now

**Review by gustavoaz7 — APPROVED** [review]

**Pull request merged**